### PR TITLE
HourlyForm Enhancements

### PR DESCRIPTION
**Pull request by Patowhiz**

### CODE REVIEW

File: ClimsoftVer4/ClimsoftVer4/ucrHourly.vb [hunk 1]
```diff
@@ -147,87 +118,52 @@ Public Class ucrHourly
             CallByName(fhRecord, kvpTemp.Value.Value.GetField(), CallType.Set, kvpTemp.Key.GetValue)
         Next
         ucrLinkedNavigation.UpdateNavigationByKeyControls()
+        'change the validation of the controls
         SetValueValidation()
     End Sub
+
+    Public Sub SaveRecord()
```
Comment (maxwellfundi) [inline ClimsoftVer4/ClimsoftVer4/ucrHourly.vb:125]: I see you like having these subs here other than running them on the dialog, why is that?

**Comment by Patowhiz** [comment]
Cause the big control is the record. This also follows the principles of
encapsulation. The original idea was to have the controls to know where to
write their values. In this case, our big controls, have knowledge of the
table and fields. The big control will also know when to update and when to
save. That's why we are also giving it a navigation control. It seems
logical to me to have that inside it. I've also noticed the disabling based
on the hour is being done on the form and not the control. This can also be
done inside the control

**Comment by maxwellfundi** [comment]
@Patowhiz  Am merging this now.

**Comment by Patowhiz** [comment]
I was to add the computed total to the total text box

**Pull request merged**